crates/rpc/rpc-types/src/eth/trace/geth/mod.rs

```diff
@@ -313,6 +313,58 @@ pub struct GethDefaultTracingOptions {
 }
 
 impl GethDefaultTracingOptions {
+    /// build a new GethDebugTracingOptions with default value
+    pub fn new() -> Self {
+        GethDefaultTracingOptions::default()
+    }
+
+    /// Sets the enable_memory field.
+    pub fn with_enable_memory(mut self, enable: bool) -> Self {
+        self.enable_memory = Some(enable);
+        self
+    }
+
+    /// Sets the disable_memory field.
+    pub fn with_disable_memory(mut self, disable: bool) -> Self {
+        self.disable_memory = Some(disable);
+        self
+    }
+
+    /// Sets the disable_stack field.
+    pub fn with_disable_stack(mut self, disable: bool) -> Self {
+        self.disable_stack = Some(disable);
+        self
+    }
+
+    /// Sets the disable_storage field.
+    pub fn with_disable_storage(mut self, disable: bool) -> Self {
+        self.disable_storage = Some(disable);
+        self
+    }
+
+    /// Sets the enable_return_data field.
+    pub fn with_enable_return_data(mut self, enable: bool) -> Self {
+        self.enable_return_data = Some(enable);
+        self
+    }
```

Comment:
these are are great, but we also want additional functions that set all fo these to true

like `enable_memory() {self.with_enabled_memory(true)}`

because the default of all of these is usually false, so it's more useful to have a function that simply flips the default

Reply:
please let me know something. You want a function that set all elements to true or a function for each elements that are true ? 

because right now we can do 

```rust
let example =  GethDefaultTracingOptions::new()
    .with_enable_memory(true)
    .with_disable_stack(true)
    .with_disable_storage(false)
    .with_enable_return_data(true)
    .with_debug(false)
    .with_limit(1000);
   ```

Reply:
> these are are great, but we also want additional functions that set all fo these to true
> 
> like `enable_memory() {self.with_enabled_memory(true)}`
> 
> because the default of all of these is usually false, so it's more useful to have a function that simply flips the default

i added function now we can do : 

```rust

let options = GethDefaultTracingOptions::enable_memory()
    .disable_stack()
    .enable_return_data();
```